backend/drtrottoir/views/schedule_definition.py

```diff
@@ -88,11 +92,40 @@ class ScheduleDefinitionViewSet(
     @action(detail=True)
     def buildings(self, request, pk=None):
         schedule_definition = self.get_object()
-        buildings = schedule_definition.buildings.all()
-        serializer = BuildingSerializer(buildings, many=True)
+        buildings = ScheduleDefinitionBuilding.objects.filter(
+            schedule_definition=schedule_definition
+        ).order_by("position")
+        serializer = ScheduleDefinitionBuildingSerializer(buildings, many=True)
 
         return Response(serializer.data)
 
+    # This counts as a different action than 'buildings', and therefore falls
+    # under the default permission_classes
+    @buildings.mapping.post
+    def set_buildings(self, request, pk=None):
+        serializer = ScheduleDefinitionBuildingSerializer(data=request.data, many=True)
+        serializer.is_valid(raise_exception=True)
+
+        schedule_definition = self.get_object()
+
+        # Remove old order & replace with new order
+        ScheduleDefinitionBuilding.objects.filter(
+            schedule_definition=schedule_definition
+        ).delete()
+        ScheduleDefinitionBuilding.objects.bulk_create(
+            (
+                ScheduleDefinitionBuilding(
+                    schedule_definition=schedule_definition,
+                    building=binding["building"],
+                    position=binding["position"],
+                )
+                for binding in serializer.validated_data
```

Comment:
Is binding not supposed to be named building here, or does the word binding have a special meaning in this case?